bluesky_queueserver/manager/comms.py

```diff
@@ -102,3 +189,233 @@ def _conn_received(self, msg):
         if response:
             response = response.json
             self._conn.send(response)
+
+
+class PipeJsonRpcSendAsync:
+    """
+    The class contains functions for supporting asyncio-based client for JSON RPC comminucation
+    using interprocess communication pipe. The class object must be created on the loop (from one of
+    `async` functions). This implementation allows calls only to one method at a time. Multiple
+    `send_msg` requests may be put on the loop, but the next message is never sent before
+    the response to the previous message is received or timeout occurred.
+
+    Parameters
+    ----------
+    conn: multiprocessing.Connection
+        Reference to bidirectional end of a pipe (multiprocessing.Pipe)
+    timeout: float
+        Default value of timeout: maximum time to wait for response after a message is sent
+    name: str
+        Name of the receiving thread (it is better to assign meaningful unique names to threads.
+
+    Examples
+    --------
+
+    .. code-block:: python
+
+        conn1, conn2 = multiprocessing.Pipe()
+
+        async def send_messages():
+            # Must be instantiated and used within the loop
+            p_send = PipeJsonRpcSendAsync(conn=conn1, name="comm-client")
+            p_send.start()
+
+            method = "method_name"
+            params = {"value": 10}   #   or list of args [10, 25]
+            response = await p_send.send_msg(method, params, notification=notification)
+
+            p_send.stop()
+
+        asyncio.run(send_messages())
+        pc.stop()
+
+
+        pc = PipeJsonRpcSendAsync(conn=conn1, name="RE QServer Receive")
+
+        def func():
+            print("Testing")
+
+        pc.add_handler(func, "some_method")
+        pc.start()   # Wait and process commands
+        # The function 'func' is called when the message with method=="some_method" is received
+        pc.stop()  # Stop before exit to stop the thread.
+
+    """
+    def __init__(self, conn, *, timeout=0.5, name="RE QServer Comm"):
+        self._conn = conn
+        self._loop = self._loop = asyncio.get_running_loop()
+
+        self._thread_name = name
+
+        self._fut_comm = None  # Future for waiting for messages from watchdog
+        self._event_comm = asyncio.Event()  # Event which is set when message is expected from watchdog
```

Comment:
I don't think we need this Event.  We have the lock to make sure we only ever have one "send" in flight at a time and the event loop will make sure the async tasks don't step on each other's toes, and the `_expected_msg_id` attribute is enough to tell us "we are expecting a message back or not".